8273655: content-types.properties files are missing some common types

FrauBoes opened the pull request:
This change adds some common types to the content-type.properties files, notably .js, .css, and .jar, as well as some others. 

The duplicated entry for .zip is removed from the Windows properties file.

<!-- Anything below this marker will be automatically updated, please do not edit manually! -->
---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8273655](https://bugs.openjdk.java.net/browse/JDK-8273655): content-types.properties files are missing some common types


### Reviewers
 * [Brian Burkhalter](https://openjdk.java.net/census#bpb) (@bplb - **Reviewer**)
 * [Daniel Fuchs](https://openjdk.java.net/census#dfuchs) (@dfuch - **Reviewer**)


### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/5506/head:pull/5506` \
`$ git checkout pull/5506`

Update a local copy of the PR: \
`$ git checkout pull/5506` \
`$ git pull https://git.openjdk.java.net/jdk pull/5506/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 5506`

View PR using the GUI difftool: \
`$ git pr show -t 5506`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/5506.diff">https://git.openjdk.java.net/jdk/pull/5506.diff</a>

</details>

anthonyvdotbe commented on src/java.base/unix/classes/sun/net/www/content-types.properties:
```diff
@@ -311,3 +319,59 @@ message/rfc822: \
 	description=XML document;\
 	file_extensions=.xml
+
+application/rtf: \
```
I'd also add an entry for `text/markdown`

FrauBoes commented:
> _Mailing list message from [Bernd Eckenfels](mailto:ecki@zusammenkunft.net) on [net-dev](mailto:net-dev@mail.openjdk.java.net):_
> 
> In the Test List .xml is not mentioned, that looks like a major format.
> 
> However I am not sure if it maybe needs extra handling because of possible different mime types for application/xml and text/xml?

.xml is already contained in the properties files as such:

```
application/xml: \
    description=XML document;\
    file_extensions=.xml
```
application/xml seems to be the default type as per RFC7303 [1], I think it should suffice in this context. 

Only the new types are added to the test. 

[1] https://datatracker.ietf.org/doc/html/rfc7303#section-4.1

FrauBoes commented:
@anthonyvdotbe Thanks for the comments, I incorporated them with one exception: I didn't add typescript (.ts) as it conflicts with MPEG transport stream on several platforms. 

A note on the test: I refactored it slightly to account for platform-specific type detection, for example in the case of .js, which is mapped to `application/javascript` on linux-aarch64 rather than `text/javascript`. The reason for this deviation is that the test uses `Files.probeContentType`, which only falls back to the content-types.properties file if no file type detector is installed.

Testing: tier1-3 all clear.

prrace commented:
I don't understand why these files are platform-specific.
And if each platform port does need one, where is the one for macOS ?

FrauBoes commented:
> I don't understand why these files are platform-specific.
> And if each platform port does need one, where is the one for macOS ?

I'm not an expert in the area, it's probably down to historic reasons? The Unix version would be for all Unix varieties, unless a more specific source file is provided (thanks to @Michael-Mc-Mahon).

FrauBoes commented:
/integrate
